astropy/units/tests/test_structured.py

```diff
@@ -512,7 +512,7 @@ def test_inplace_conversion(self):
         q1 = q_pv.copy()
         q_link = q1
         q1 <<= StructuredUnit(('AU', 'AU/day'))
-        assert q1 is q_link
+        assert q1 is not q_link
```

Comment:
Maybe add a note here that `# In principle, in-place might be possible, in which case this should be changed.`

Or should we just remove the test if we do not care about whether it is in-place??

Reply:
I'm conflicted on this. The memory share is a speed-up, but if it is only sometimes used then it can get pretty confusing. This is the current situation as well, this PR just changes when. If we're optimizing for intuitiveness, then I think we should get rid of memory-share. If we're going for speed, then memory-share should be kept wherever possible.

Reply:
> Anyway, if not removing the assert, I do think we should have a comment, so that if it is changed in future, there is no worry about changing the test.

Agreed

Reply:
✅

Reply:
✅